src/core/default_appearance.js

```diff
@@ -87,6 +87,83 @@ function parseDefaultAppearance(str) {
   return new DefaultAppearanceEvaluator(str).parse();
 }
 
+class AppearanceStreamEvaluator extends EvaluatorPreprocessor {
+  constructor(stream) {
+    super(stream);
+    this.stream = stream;
+  }
+
+  parse() {
+    const operation = {
+      fn: 0,
+      args: [],
+    };
+    const result = {
+      fontSize: 0,
+      fontName: "",
+      fontColor: /* black = */ new Uint8ClampedArray(3),
+    };
+
+    let parsingText = false;
+    let scaleFactor = 1;
+
+    try {
+      while (true) {
+        operation.args.length = 0; // Ensure that `args` it's always reset.
+
+        if (!this.read(operation)) {
+          break;
+        }
+        const { fn, args } = operation;
+
+        switch (fn | 0) {
+          case OPS.beginText:
+            parsingText = true;
+            break;
+          case OPS.endText:
+            parsingText = false;
+            scaleFactor = 1;
+            break;
+          case OPS.setTextMatrix:
+            scaleFactor *= Math.hypot(args[0], args[1]);
+            break;
+          case OPS.setFont:
+            if (!parsingText) {
+              break;
+            }
```

Comment:
Do we actually want to ignore setFont *outside* of the beginText/endText operators, since we don't do that in `DefaultAppearanceEvaluator` above?

Reply:
The DA isn't really supposed to contain BT/ET but just the minimal commands to set the style.
It probably doesn't hurt to get the font info wherever they're.